Objective-C/CBLBlob.mm

```diff
@@ -367,7 +367,7 @@ - (void) fl_encodeToFLEncoder: (FLEncoder)encoder {
         }
     }
     
-    NSDictionary* dict = self.jsonRepresentation;
+    NSDictionary* dict = FLEncoder_isFleece(encoder) ? self.jsonRepresentation : self.properties;
```

Comment:
https://github.com/couchbaselabs/fleece/pull/87

Reply:
I think we should take more direct approach here as it's impossible to understand when reading this code only why FleeceEncoder uses data from json representation and why JSONEncoder uses properties. I suggested that we create a new FLEncoderContext that can be used to put into the FLEncoder's extra info. FLEncoderContext could have two properties:
* Nullable document.
* boolean encodeQueryParameter (so that we really know what is going on here).

Reply:
I think we should rename jsonRepresentation to something like

- (NSDictionary*) blobProperties: (BOOL)mayIncludeContent;

Reply:
Updated the code with FLEncoderContext.